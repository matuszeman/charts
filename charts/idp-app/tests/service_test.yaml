suite: service
templates:
  - service.yaml
values:
  - values/minimum.yaml
tests:
  - it: should match snapshot with values-ci.yaml
    values:
      - ../values-ci.yaml
    asserts:
      - matchSnapshot: {}
  - it: when service is enabled but not deployment
    values:
      - values/service.yaml
    asserts:
      - failedTemplate:
          errorMessage: "service: deployment must be enabled"
  - it: when multi deployment
    values:
      - values/service.yaml
      - values/deployment.yaml
      - values/multi.yaml

    asserts:
      - matchSnapshot: {}
      - equal:
          path: metadata.name
          value: RELEASE-NAME-one
        documentIndex: 0
      - equal:
          path: spec.selector["app.kubernetes.io/instance"]
          value: RELEASE-NAME-one
        documentIndex: 0
      - equal:
          path: metadata.name
          value: RELEASE-NAME-two
        documentIndex: 1
      - equal:
          path: spec.selector["app.kubernetes.io/instance"]
          value: RELEASE-NAME-two
        documentIndex: 1

  - it: when targetPortProtocol is defined
    values:
      - values/service.yaml
      - values/deployment.yaml
    set:
      service:
        targetPortProtocol: UDP
    asserts:
      - equal:
          path: spec.ports[0].protocol
          value: UDP
