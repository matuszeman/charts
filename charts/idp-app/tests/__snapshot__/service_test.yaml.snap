should match snapshot with values-ci.yaml:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: apache
        - name: anotherport
          port: 9000
          targetPort: anotherport
      selector:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/name: app-name
      type: ClusterIP
when multi deployment:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-one
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: app
      selector:
        app.kubernetes.io/instance: RELEASE-NAME-one
        app.kubernetes.io/name: app-name
      type: ClusterIP
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-two
    spec:
      ports:
        - name: app
          port: 80
          protocol: TCP
          targetPort: app
      selector:
        app.kubernetes.io/instance: RELEASE-NAME-two
        app.kubernetes.io/name: app-name
      type: ClusterIP
